contracts/src/main/kotlin/com/contractsAndStates/states/GameBoardState.kt

```diff
@@ -55,7 +56,7 @@ data class GameBoardState(val hexTiles: PlacedHexTiles,
 }
 
 @CordaSerializable
-data class PlacedHexTiles(val value: List<HexTile>) {
+data class PlacedHexTiles(val value: MutableList<HexTile>) {
```

Comment:
@xavierlepretre - had to revert to MutableList for several data types inside of HexTile. I was getting a serialization error related to using List here. 

I know this undermines the point of rebuilding the GameBoardState (which looks absolutely amazing by the way). Happy to debug this together on a seperate branch.